crypto/evp/p_lib.c

```diff
@@ -318,8 +319,13 @@ EVP_PKEY *EVP_PKEY_new_CMAC_key(ENGINE *e, const unsigned char *priv,
                                 size_t len, const EVP_CIPHER *cipher)
 {
 #ifndef OPENSSL_NO_CMAC
+    const char *engine_name = e != NULL ? ENGINE_get_name(e) : NULL;
+    const char *cipher_name = EVP_CIPHER_name(cipher);
     EVP_PKEY *ret = EVP_PKEY_new();
-    EVP_MAC_CTX *cmctx = EVP_MAC_CTX_new_id(EVP_MAC_CMAC);
+    EVP_MAC *cmac = EVP_MAC_fetch(NULL, "CMAC", NULL);
```

Comment:
Should we use the same libctx that the cipher was fetched from (if it was)?

Reply:
That would be sensible

Reply:
Done, will show up in a commit soon

Reply:
[In the autumn?](https://gist.github.com/levitte/5cbabd94b19f1a022fdd1c2f4442e2b7)

...

or I can have a look now...

Reply:
Done